### ISSUE TITLE
Warn when JSDoc type cast misses parentheses

### ISSUE THREAD

**Issue by saschanaz** [Microsoft/TypeScript#28993]
## Search Terms

jsdoc. type cast

## Suggestion

JSDoc type cast currently requires parentheses for technical reasons (#18212), but it does not warn when they don't exist.

## Use Cases

```js
// This doesn't work, the JSDoc comment is silently ignored
// A warning message will help.
const img = /** @type {HTMLImageElement} */ document.getElementById('#cat')
img.src = 'cat.gif'
```

## Examples

```js
const img = /** @type {HTMLImageElement} */ document.getElementById('#cat')
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            ^ Warning: The JSDoc type cast misses parentheses and is being ignored
img.src = 'cat.gif'
```

## Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by DanielRosenwasser** [comment]
I'd think that for the same reason that it's expensive to try to look for JSDoc comments for type-assertions to apply, it's expensive to detect when you forgot parentheses.

But this doesn't have to be that expensive? You could do a text search for `@type` and try to find locations that are otherwise not affected. What do you think @sandersn?
